openstates/de/bills.py

```diff
@@ -1,693 +1,339 @@
-        # window-level vars named "moniker" and "docnum".
-        if docname == "Fiscal Note":
-            xpath = '//script[contains(., "var Fdocnum")]'
         else:
-            xpath = '//script[contains(., "var docnum")]'
-        script_text = _doc.xpath(xpath)[0].text_content()
-        docnum = re_docnum.search(script_text).group(1)
-        moniker = re_moniker.search(script_text).group(1)
+            bill = Bill(session,chamber,bill_id,bill_title)
 
-        # Mimetypes.
-        formats = ['.html', '.pdf', '.docx', '.Docx']
-        mimetypes = {
-            '.html': 'text/html',
-            '.pdf': 'application/pdf',
-            '.docx': 'application/msword'
-            }
+            for s in primary_sponsors:
+                bill.add_sponsor("primary",s)
 
-        for format_ in formats:
+            for s in addl_sponsors:
+                #it is not totally clear whether "additional sponsors"
+                #are co or primary but primary is my best guess
+                #based on the bill text, bc they're on the first
+                #line with the primary sponsor
+                bill.add_sponsor("primary",s)
 
-            el =_doc.xpath('//font[contains(., "%s%s")]' % (filename, format_))
+            for s in cosponsors:
+                bill.add_sponsor("cosponsor",s)
+
+        for name, doc_link in bill_documents.items():
+            if "Engrossment" in name or "Bill Text" in name:
+                if doc_link not in sub_versions:
+                    bill.add_version(name,doc_link,mimetype="text/html")
+            else:
+                pass
+                bill.add_document(name,doc_link,mimetype="text/html")
+
+        for a in action_list:
+            if a.strip():
+                date, action = a.split('-', 1)
+                try:
+                    date = datetime.strptime(date.strip(), '%b %d, %Y')
+                except ValueError:
+                    date = datetime.strptime(date.strip(), '%B %d, %Y') # XXX: ugh.
+                action = action.strip()
+                actor = actions.get_actor(action, bill['chamber'])
+                attrs = dict(actor=actor, action=action, date=date)
+                attrs.update(**self.categorizer.categorize(action))
+                bill.add_action(**attrs)
+
+        for name, doc in vote_documents.items():
+            vote_chamber = "lower" if "house" in name.lower() else "upper"
+            vote_page = self.lxmlize(doc)
+            vote_info = vote_page.xpath(".//div[@id='page_content']/p")[-1]
+            yes_votes = []
+            no_votes = []
+            other_votes = []                        
+            lines = vote_info.text_content().split("\n")
+            for line in lines:
+                if line.strip().startswith("Date"):
+                    date_str = " ".join(line.split()[1:4])
+                    date = datetime.strptime(date_str,"%m/%d/%Y %I:%M %p")
+                    passed = "Passed" in line
```

Comment:
Is "Passed" the only keyword for passage? From the pages I've looked at it is, but maybe we could check the last bold word in `line`, and `assert` that it's in a list of accepted statuses?